init.c

```diff
@@ -694,27 +694,38 @@ static int cutorch_setKernelPeerToPeerAccess(lua_State *L)
 }
 
 static int cutorch_getMemoryUsage(lua_State *L) {
-  size_t freeBytes = 0;
   size_t totalBytes = 0;
-  int curDevice;
-  THCudaCheck(cudaGetDevice(&curDevice));
+  size_t freeBytes = 0;
+
+  THCState *state = cutorch_getstate(L);
 
   int device = luaL_optint(L, 1, -10);
-  if (device == -10) { /* no argument passed, current device mem usage */
-    THCudaCheck(cudaMemGetInfo(&freeBytes, &totalBytes));
-  } else { /* argument was given, particular device's memory usage */
-    THCudaCheck(cudaSetDevice(device-1)); /* zero indexed */
-    THCudaCheck(cudaMemGetInfo(&freeBytes, &totalBytes));
```

Comment:
Right, me trying to avoid code duplication here led to code bloat instead :). Cleaned up.